### PR TITLE
🎥 ❌ CLM-76 Enforce Whatsapp Media Limit

### PR DESCRIPTION
**Pull request by LemmyMwaura**
# Description.
Enforce Whatsapp media limit on all media blocks before upload.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: apps/conv-learning-manager/src/assets/i18n/en.json [hunk 1]
```diff
@@ -181,6 +181,27 @@
           "NAME": "Give This media Item a Name",
           "UPDATE": "Update Video"
         }
+      }, 
+      "SIZE-LIMIT-MESSAGE": {
+        "AUDIO": {
+          "MESSAGE-ONE": "Audio cannot be sent to whatsapp,",
```
Comment (ReaganCn) [inline apps/conv-learning-manager/src/assets/i18n/en.json:187]: You can group whatsapp messages together through a property "WHATSAPP", to be able to accomodate other platform requirements later

File: libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.html [hunk 2]
```diff
@@ -18,6 +18,12 @@
                 <input #audioInput (change)="processAudio($event)" type="file" accept=".mp3" class="fileSrc">
             </div>
         </div>
+
+        <div class="error-message" *ngIf="byPassedLimits.length">
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.html:22]: This should be an array. For now there should also be one item in the array

File: libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts [hunk 3]
```diff
@@ -1,51 +1,73 @@
...
-    //Step 1 - Create the file path that will be in firebase storage
-    const audioFilePath = `audios/${this.file.name}_${new Date().getTime()}`;
-    (await this._audioUploadService.uploadFile(this.file, this.block, audioFilePath)).subscribe();
-  }
-}
+import { Component, OnInit, Input, OnDestroy } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+
+import { take } from 'rxjs';
+import { SubSink } from 'subsink';
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts:5]: subsink before rxjs

File: libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts [hunk 4]
```diff
@@ -1,51 +1,73 @@
...
+    if (this.file) {
+      this.isLoadingAudio = true;
+
+      //Step 1 - Create the file path that will be in firebase storage
+      const audioFilePath = `audios/${this.file.name}_${new Date().getTime()}`;
+      
+      //Step 2 - Upload file to firestore
+      const response = await this._audioUploadService.uploadSingleFile(this.file, audioFilePath);
+
+      //Step 3 Check if file bypasses size limit.
```
Comment (ReaganCn) [inline libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts:52]: Comment is wrongly placed

**Review by ReaganCn — CHANGES_REQUESTED** [review]
Just some minor changes

File: libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.html (hunk 2 above)
Comment (LemmyMwaura) [reply libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.html:22]: It's an array.

File: libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts (hunk 4 above)
Comment (LemmyMwaura) [reply libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts:52]: Moved

File: libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts (hunk 3 above)
Comment (LemmyMwaura) [reply libs/features/convs-mgr/stories/blocks/library/audio-message-block/src/lib/components/audio-block/audio-block.component.ts:5]: ?? any reason why. They are both third-party modules.

**Review by ReaganCn — APPROVED** [review]

**Pull request merged**